addons/mail/static/tests/mail_test_helpers.js

```diff
@@ -284,6 +284,7 @@ export async function start(options) {
     } else {
         env = getMockEnv() || (await makeMockEnv());
     }
+    env.testEnv = true;
```

Comment:
This is fine, could be even better with Symbol to ensure no conflicts (and no need to comment in contains helpers code):

```is
export IS_TEST_ENV = Symbol("isTestEnv");
env[IS_TEST_ENV] = true;
if (env?.[IS_TEST_ENV]) {
   target = env.target;
}
```